torch/linalg/__init__.py

```diff
@@ -204,31 +213,42 @@
 slogdet = _add_docstr(_linalg.linalg_slogdet, r"""
 linalg.slogdet(input, *, out=None) -> (Tensor, Tensor)
 
-Calculates the sign and natural logarithm of the absolute value of a square matrix's determinant,
-or of the absolute values of the determinants of a batch of square matrices :attr:`input`.
-The determinant can be computed with ``sign * exp(logabsdet)``.
+Computes the sign and logarithm of the norm of the determinant of a square matrix or batch of square matrices.
 
-Supports input of float, double, cfloat and cdouble datatypes.
+For a complex :attr:`input`, as the determinant is complex in this case, it returns the angle
+and the logarithm of the modulus of the determinant.
 
-.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+The returend value decomposition is represented as a namedtuple `(sign, logabsdet)`.
+If :attr:`input` is a batch of matrices, then `sign`, `logabsdet` are also batched with the same
+batch dimensions. The determinant can be recovered as `sign * exp(logabsdet)`.
 
-.. note:: The determinant is computed using LU factorization. LAPACK's `getrf` is used for CPU inputs,
-          and MAGMA's `getrf` is used for CUDA inputs.
+Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
+`logabsdet` will always be real-valued, even if :attr:`input` is complex.
+`sign` will have the same dtype as :attr:`input`.
 
-.. note:: For matrices that have zero determinant, this returns ``(0, -inf)``.
-          If :attr:`input` is batched then the entries in the result tensors corresponding to matrices with
-          the zero determinant have sign 0 and the natural logarithm of the absolute value of the determinant -inf.
+.. note:: When :attr:`input` is on a CUDA device, this function synchronizes that device with the CPU.
+
+.. note:: This function is computed using :func:`torch.lu`.
```

Comment:
This note can probably be combined with the previous note. In the future we'll have an "Implementation details" section of the docs, so everything we'd put into that section can go into the same note.

Reply:
This is a great point! Shorter is definitely better.

I have merged the implementation with the synchronisation in all the docs. The "implementation details" section would be a great one, certainly.